src/library/features/analysis/analysisfeature.cpp

```diff
@@ -153,32 +163,55 @@ void AnalysisFeature::slotProgressUpdate(int num_left) {
 
 void AnalysisFeature::stopAnalysis() {
     //qDebug() << this << "stopAnalysis()";
-    if (m_pAnalyzerQueue != NULL) {
+    if (m_pAnalyzerQueue != nullptr) {
         m_pAnalyzerQueue->stop();
     }
 }
 
 void AnalysisFeature::cleanupAnalyzer() {
     setTitleDefault();
     emit(analysisActive(false));
-    if (m_pAnalyzerQueue != NULL) {
+    if (m_pAnalyzerQueue != nullptr) {
         m_pAnalyzerQueue->stop();
         m_pAnalyzerQueue->deleteLater();
-        m_pAnalyzerQueue = NULL;
+        m_pAnalyzerQueue = nullptr;
         // Restore old BPM detection setting for preferences...
         m_pConfig->set(ConfigKey("[BPM]","BPMDetectionEnabled"), ConfigValue(m_iOldBpmEnabled));
     }
 }
 
+void AnalysisFeature::tableSelectionChanged(const QItemSelection&,
+                                            const QItemSelection&) {
+    //qDebug() << "AnalysisFeature::tableSelectionChanged" << sender();
+    QPointer<WTrackTableView> pTable = LibraryFeature::getFocusedTable();
+    if (pTable.isNull()) {
+        return;
+    }
+    
+    QModelIndexList indexes = pTable->selectionModel()->selectedIndexes();
+    m_pAnalysisView->setSelectedIndexes(indexes);
+}
+
 bool AnalysisFeature::dropAccept(QList<QUrl> urls, QObject* pSource) {
     Q_UNUSED(pSource);
-    QList<QFileInfo> files = DragAndDropHelper::supportedTracksFromUrls(urls, false, true);
+    QList<QFileInfo> files =
+            DragAndDropHelper::supportedTracksFromUrls(urls, false, true);
     // Adds track, does not insert duplicates, handles unremoving logic.
-    QList<TrackId> trackIds = m_pTrackCollection->getTrackDAO().addMultipleTracks(files, true);
+    QList<TrackId> trackIds =
+            m_pTrackCollection->getTrackDAO().addMultipleTracks(files, true);
     analyzeTracks(trackIds);
     return trackIds.size() > 0;
 }
 
 bool AnalysisFeature::dragMoveAccept(QUrl url) {
     return SoundSourceProxy::isUrlSupported(url);
 }
+
+AnalysisLibraryTableModel* AnalysisFeature::getAnalysisTableModel() {
```

Comment:
Mmm in this case this can be created in the constructor and we can remove the pointer.